frappe/share.py

```diff
@@ -65,22 +65,25 @@ def remove(doctype, name, user, flags=None):
 
 
 @frappe.whitelist()
-def set_permission(doctype, name, user, permission_to, value=1, everyone=0):
+def set_permission(doctype, name, user, permission_to, value=1, everyone=0, flags=None):
```

Comment:
This is a bad idea. You're basically accepting user input when it's called from the client side. (Note that this function is whitelisted)

Reply:
Hi @ankush, thanks for your quick feedback. Since we have the same here:
<img width="730" alt="Bildschirmfoto 2022-10-14 um 13 58 10" src="https://user-images.githubusercontent.com/49870752/195841635-365af61e-562a-4866-921e-ff4043ad2b0f.png">
would it make sense to "hide" the function which actually creates the docshare in `add` and `set_permission` one level deeper (which then accepts flags) while the whitelisted methods don't?

Reply:
Hi @ankush I did some changes. Is this sufficient or would you refactor it differently?